Mathlib/Algebra/Category/GroupCat/Abelian.lean

```diff
@@ -47,18 +45,42 @@ def normalEpi (_ : Epi f) : NormalEpi f :=
 set_option linter.uppercaseLean3 false in
 #align AddCommGroup.normal_epi AddCommGroupCat.normalEpi
 
-end
-
 /-- The category of abelian groups is abelian. -/
 instance : Abelian AddCommGroupCat.{u} where
-  has_finite_products := ⟨by infer_instance⟩
+  has_finite_products := ⟨HasFiniteProducts.out⟩
   normalMonoOfMono := normalMono
   normalEpiOfEpi := normalEpi
-  add_comp := by
-    intros
-    simp only [Preadditive.add_comp]
-  comp_add := by
-    intros
-    simp only [Preadditive.comp_add]
+  add_comp := by exact Preadditive.add_comp
+  comp_add := by exact Preadditive.comp_add
+
+theorem exact_iff : Exact f g ↔ f.range = g.ker := by
+  rw [Abelian.exact_iff' f g (kernelIsLimit _) (cokernelIsColimit _)]
+  exact
+    ⟨fun h => ((range_le_ker_iff _ _).mpr h.left).antisymm ((ker_le_range_iff _ _).mpr h.right),
+      fun h => ⟨(range_le_ker_iff _ _).mp h.le, (ker_le_range_iff _ _).mp h.symm.le⟩⟩
+
+theorem exact_iff' : Exact f g ↔ f ≫ g = 0 ∧ g.ker ≤ f.range := by
```

Comment:
You could think about turning this into a lemma `range_eq_ker_iff` that lives next to `range_le_ker_iff`.

Reply:
You mean `exact_iff`? This name is analogous to the one in `ModuleCat`, so if we'd have to change both.

Reply:
Sorry, I guess what I wanted to say is that the lemma `exact_iff'` doesn't really have any content. Indeed, if you add the missing
```lean
theorem comp_def : f ≫ g = g.comp f :=
  rfl
```
to `GroupCat/Basic.lean`, then `exact_iff'` is just
```lean
theorem exact_iff' : Exact f g ↔ f ≫ g = 0 ∧ g.ker ≤ f.range := by
  rw [exact_iff, le_antisymm_iff, AddMonoidHom.range_le_ker_iff, comp_def]
```
You can keep it with the shortened proof if you like, but you could also just inline it into the instance below.